runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -121,33 +121,56 @@ JITServerHelpers::insertIntoOOSequenceEntryList(ClientSessionData *clientData, T
       {
-      if (JITServerHelpers::serverMsgTypeCount[i] > 0)
+      for (int i = 0; i < JITServer::MessageType_ARRAYSIZE; ++i)
          {
-         j9tty_printf(PORTLIB, "#%04d %7u", i, JITServerHelpers::serverMsgTypeCount[i]);
+         if (JITServerHelpers::serverMsgTypeCount[i] > 0)
+            {
+            j9tty_printf(PORTLIB, "#%04d %7u", i, JITServerHelpers::serverMsgTypeCount[i]);
 #ifdef MESSAGE_SIZE_STATS            
-         j9tty_printf(PORTLIB, "\t%f\t%f\t%f\t%f\t%f", JITServer::CommunicationStream::collectMsgStat[i].maxVal(), 
+            j9tty_printf(PORTLIB, "\t%f\t%f\t%f\t%f\t%f", JITServer::CommunicationStream::collectMsgStat[i].maxVal(), 
                      JITServer::CommunicationStream::collectMsgStat[i].minVal(), JITServer::CommunicationStream::collectMsgStat[i].mean(),
                      JITServer::CommunicationStream::collectMsgStat[i].stddev(), JITServer::CommunicationStream::collectMsgStat[i].sum());
 #endif
-         j9tty_printf(PORTLIB, "\t\t%s\n", JITServer::messageNames[i]);
-         totalMsgCount += JITServerHelpers::serverMsgTypeCount[i];
+            j9tty_printf(PORTLIB, "\t\t%s\n", JITServer::messageNames[i]);
+            totalMsgCount += JITServerHelpers::serverMsgTypeCount[i];
+            }
          }
+      if (JITServerHelpers::serverMsgTypeCount[0])
+         j9tty_printf(PORTLIB, "Total number of messages: %d. Average number of messages per compilation:%f\n", totalMsgCount, totalMsgCount/float(JITServerHelpers::serverMsgTypeCount[0]));
       }
-   if (JITServerHelpers::serverMsgTypeCount[0])
-       j9tty_printf(PORTLIB, "Total number of messages: %d. Average number of messages per compilation:%f\n", totalMsgCount, totalMsgCount/float(JITServerHelpers::serverMsgTypeCount[0]));
+   else if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
+      {
+      // to print in server, run ./jitserver -Xdump:jit:events=user
+      // then kill -3 <pidof jitserver>
+      for (int i = 0; i < JITServer::MessageType_ARRAYSIZE; ++i)
+         {
+#ifdef MESSAGE_SIZE_STATS  
+         if (JITServer::CommunicationStream::collectMsgStat[i].samples() > 0)
+            { 
+            j9tty_printf(PORTLIB, "#%04d %7u", i, JITServer::CommunicationStream::collectMsgStat[i].samples());            
+            j9tty_printf(PORTLIB, "\t%f\t%f\t%f\t%f\t%f", JITServer::CommunicationStream::collectMsgStat[i].maxVal(), 
+                     JITServer::CommunicationStream::collectMsgStat[i].minVal(), JITServer::CommunicationStream::collectMsgStat[i].mean(),
+                     JITServer::CommunicationStream::collectMsgStat[i].stddev(), JITServer::CommunicationStream::collectMsgStat[i].sum());
 
+            j9tty_printf(PORTLIB, "\t\t%s\n", JITServer::messageNames[i]);
+            totalMsgCount += JITServer::CommunicationStream::collectMsgStat[i].samples();
+#endif
+            }
+         }
+      j9tty_printf(PORTLIB, "Total number of messages: %u\n", totalMsgCount);
```

Comment:
If `MESSAGE_SIZE_STATS` is not defined, the `for` loop above will go over all the messages, but print nothing and at the very end it will print "Total number of messages: 0"
I would include the `for` loop and this last printf message into the block protected by ifdef.
Also, add a comment to the `#endif` so that we know what should be paired with:

```
#if defined(MESSAGE_SIZE_STATS)
...
#endif // defined(MESSAGE_SIZE_STATS)